gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceSource.java

```diff
@@ -155,19 +133,25 @@ protected void addLineageSourceInfo(SourceState sourceState, SourceEntity entity
   }
   @Override
   protected List<WorkUnit> generateWorkUnits(SourceEntity sourceEntity, SourceState state, long previousWatermark) {
-    List<WorkUnit> workUnits = null;
-    workUnitConf = new SfConfig(state.getProperties());
+    SalesforceConnector connector = getConnector(state);
```

Comment:
if used only to initialize the histogram service, scope it to within that path of the conditional

Reply:
This needs to update the class variable. Updated, thanks.